src/FluxCalculator.jl

```diff
@@ -91,4 +100,296 @@ function calculate_surface_air_density(atmos_sim::Interfacer.AtmosModelSimulatio
+    FT = eltype(csf[1])
+
+    # reset coupler fields
+    csf.F_turb_ρτxz .*= FT(0)
+    csf.F_turb_ρτyz .*= FT(0)
+    csf.F_turb_energy .*= FT(0)
+    csf.F_turb_moisture .*= FT(0)
+
+    # iterate over all columns (when regridding, this will need to change)
+    Fields.bycolumn(boundary_space) do colidx
+        # atmos state of center level 1
+        z_int = Interfacer.get_field(atmos_sim, Val(:height_int), colidx)
+        uₕ_int = Interfacer.get_field(atmos_sim, Val(:uv_int), colidx)
+        thermo_state_int = Interfacer.get_field(atmos_sim, Val(:thermo_state_int), colidx)
+        z_sfc = Interfacer.get_field(atmos_sim, Val(:height_sfc), colidx)
+
+        for sim in model_sims
+            # iterate over all surface models with non-zero area fractions
+            if sim isa Interfacer.SurfaceModelSimulation
+                area_fraction = Interfacer.get_field(sim, Val(:area_fraction), colidx)
+                area_mask = Regridder.binary_mask.(area_fraction, threshold = eps())
+
+                if !iszero(parent(area_mask))
+
+                    thermo_state_sfc = surface_thermo_state(sim, thermo_params, thermo_state_int, colidx)
+
+                    # set inputs based on whether the surface_scheme is MOST or bulk
+                    inputs = surface_inputs(
+                        surface_scheme,
+                        thermo_state_sfc,
+                        thermo_state_int,
+                        uₕ_int,
+                        z_int,
+                        z_sfc,
+                        get_scheme_specific_properties(surface_scheme, sim, colidx)...,
+                    )
+
+                    # update fluxes in the coupler
+                    surface_params = get_surface_params(atmos_sim)
+                    F_turb_ρτxz, F_turb_ρτyz, F_shf, F_lhf, F_turb_moisture =
+                        get_surface_fluxes_point!(inputs, surface_params)
+
+                    fields = (;
+                        F_turb_ρτxz = F_turb_ρτxz,
+                        F_turb_ρτyz = F_turb_ρτyz,
+                        F_turb_energy = F_shf .+ F_lhf,
+                        F_turb_moisture = F_turb_moisture,
+                    )
+
+                    # update the fluxes of this surface model
+                    update_turbulent_fluxes_point!(sim, fields, colidx)
+
+                    # add the flux contributing from this surface
+                    @. csf.F_turb_ρτxz[colidx] += F_turb_ρτxz * area_fraction * area_mask
+                    @. csf.F_turb_ρτyz[colidx] += F_turb_ρτyz * area_fraction * area_mask
+                    @. csf.F_turb_energy[colidx] += (F_shf .+ F_lhf) * area_fraction * area_mask
+                    @. csf.F_turb_moisture[colidx] += F_turb_moisture * area_fraction * area_mask
+
+                end
+            end
+        end
+
+    end
+
+    # TODO: add allowable bounds here, check explicitly that all fluxes are equal
+    # @assert(extrema(csf.F_turb_energy) ≈ (-1000.0, 1000.0))
```

Comment:
Was this meant to be uncommented?

Reply:
Yes, it's a placeholder, but the comment probably suffices.